tests/run-pass/fs.rs

```diff
@@ -3,6 +3,20 @@
 
 use std::fs::{File, remove_file};
 use std::io::{Read, Write, ErrorKind};
+use std::path::Path;
+
+#[cfg(target_os = "linux")]
+fn test_metadata(path: &Path, bytes: &[u8]) {
+    // Test that the file metadata is correct.
+    let metadata = std::fs::metadata(path).unwrap();
+    // `path` should point to a file.
+    assert!(metadata.is_file());
+    // The size of the file must be equal to the number of written bytes.
+    assert_eq!(bytes.len() as u64, metadata.len());
+}
```

Comment:
There are a *lot* of branches in your `statx`; how many of those does this explore? We should try to get as many of them as we can. I have no idea how else you'd make sure any of this actually works.^^

Reply:
We could test the following cases mentioned the in the "Invoking statx()" section of [`man statx`](http://man7.org/linux/man-pages/man2/statx.2.html):
- An absolute pathname: I think this is the one we are testing
- A relative pathname: This can be tested doing cwd to `/tmp/` and then using a relative path.
I'm not sure when "A directory-relative pathname" and "By file descriptor" are used.

Reply:
Please add `trace!` to each of these branches. Then you can run the test as
```
MIRI_LOG=miri::shims::fs=trace ./miri run tests/run-pass/fs.rs
```
to see which branch(es) are covered. Untested branches should at least have a FIXME in the code, but it is kind of questionable to even implement them if they are unreachable for Rust code. You could probably reach them by writing a linux-specific test that uses `libc` APIs.

Reply:
With the last commit, the `fs` test checks if the metadata for absolute and "relative to cwd" paths is generated correctly. To the best of my understanding there is no way to test the other branches from `libstd`.

I'm thinking it might be a better idea to just remove the remaining branches and throw an error for the moment. If someone needs one of them, that would be our test for that branch.